src/adapter/src/coord.rs

```diff
@@ -2672,55 +2677,119 @@ impl<S: Append + 'static> Coordinator<S> {
                 ReplicaConfig::Remote { .. } => None,
             };
-            let config =
-                concretize_replica_config(config, &self.replica_sizes, &self.availability_zones)?;
+            let replica_config = concretize_replica_config(
+                replica_config,
+                &self.replica_sizes,
+                &self.availability_zones,
+            )?;
+
+            // These are the persisted, per replica log collections
+            let log_collections =
+                if compute_instance_config.is_some() && self.enable_experimental_recorded_logs() {
+                    self.catalog.allocate_introspection_source_indexes().await
+                } else {
+                    Vec::new()
+                };
+
+            introspection_collections.extend(log_collections.iter().map(|(log, id)| {
+                (
+                    *id,
+                    CollectionDescription {
+                        desc: log.variant.desc(),
+                        ingestion: None,
+                        remote_addr: None,
+                    },
+                )
+            }));
+
             ops.push(catalog::Op::CreateComputeInstanceReplica {
                 name: replica_name,
-                config,
+                config: replica_config,
                 on_cluster_name: name.clone(),
                 logical_size,
+                log_collections,
             });
         }
+
         self.catalog_transact(Some(session), ops, |_| Ok(()))
             .await?;
+
+        let introspection_collection_ids = introspection_collections
+            .iter()
+            .map(|(id, _)| *id)
+            .collect();
+
+        self.dataflow_client
+            .storage_mut()
+            .create_collections(introspection_collections)
+            .await
+            .unwrap();
+
         let instance = self
             .catalog
             .resolve_compute_instance(&name)
-            .expect("compute instance must exist after creation");
+            .expect("compute instance must exist after creation")
+            .clone();
+
         self.dataflow_client
             .create_instance(instance.id, instance.logging.clone())
             .await;
         for (replica_id, replica) in instance.replicas_by_id.clone() {
             self.dataflow_client
-                .add_replica_to_instance(instance.id, replica_id, replica.config)
+                .add_replica_to_instance(
+                    instance.id,
+                    replica_id,
+                    replica.config,
+                    replica.log_collections,
+                )
                 .await
                 .unwrap();
         }
+
+        if let Some(c) = compute_instance_config {
+            self.initialize_compute_read_policies(
+                introspection_collection_ids,
+                instance.id(),
+                Some(c.granularity.as_millis() as u64),
```

Comment:
Maybe it's better to use `self.logical_compaction_window` here, granularity conveys how often an update is produced, and the window says how many past versions we should keep around...